controllers/serving/inference_controller.go

```diff
@@ -335,6 +430,72 @@ func (ir *InferenceReconciler) syncServiceForInference(inf *servingv1alpha1.Infe
 	return nil
 }
 
+// syncIstioLessIngressGateway sync the istio-less ingress gateway for inference service.
+func (ir *InferenceReconciler) syncIstioLessIngressGateway(inf *servingv1alpha1.Inference) error {
+	gatewayInCluster := v1.Deployment{}
+	err := ir.client.Get(context.Background(), types.NamespacedName{Namespace: inf.Namespace, Name: inf.Name}, &gatewayInCluster)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	svcExists := true
+	if err != nil && errors.IsNotFound(err) {
+		svcExists = false
+	}
+
+	gatewayLabels := map[string]string{
+		apiv1.LabelInferenceName: inf.Name,
+	}
+
+	gatewayName := getIstioLessGatewayName(inf)
+
+	replicas := int32(1)
+	selector := &metav1.LabelSelector{MatchLabels: gatewayLabels}
+	gateway := v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gatewayName,
+			Namespace: inf.Namespace,
+			Labels:    gatewayLabels,
+		},
+		Spec: v1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: selector,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      gatewayName,
+					Namespace: inf.Namespace,
+					Labels:    gatewayLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  gatewayName,
+							Image: GATEWAY_IMAGE,
+						},
+					},
+				},
+			},
+			Strategy: v1.DeploymentStrategy{Type: v1.RollingUpdateDeploymentStrategyType},
+		},
+	}
+
+	// Set owner-reference to ingress gateway.
+	err = controllerutil.SetControllerReference(inf, &gateway, ir.scheme)
+	if err != nil {
+		return err
+	}
+
+	if !svcExists {
+		return ir.client.Create(context.Background(), &gateway)
+	}
+
+	if !reflect.DeepEqual(gateway.Spec, gatewayInCluster.Spec) {
```

Comment:
I think we should not reset the spec every time,  in case we need to manually change something.

Reply:
I don’t quite understand your meaning here, the trigger condition here is manually changing the spec